### DISCUSSION TITLE
Is there any way to dont see file listing and still see pictures on layout?

### DISCUSSION THREAD

**Discussion by Necoo33** [actix/actix-web#3178] [Q&A] [answered]
Hello, i'm using actix-web 4.4 and actix-files 0.6.2. I want to serve my static files and use in html files and i also don't want them to be able to seen on mount path.

Here is my server code:

```rust

            .service(Files::new("/other", "files/other/").show_files_listing())
            .service(Files::new("/sample-files", "files/sample-files/").show_files_listing())
            .service(Files::new("/template-files", "files/template-files/").show_files_listing())

```

when i put that middlewares on my server everyone can see my static files on "/other", "/sample-files", "/template-files" routes on browser. I don't want my static files could be seen on that routes. When i use `.guard()` method and header if someone dont have that header they couldn't see for example pictures on html pages, but i only want to they're not able to see only file listing routes. Is there any way to achieve that behavior?

**Comment by robjtede** [comment, marked as answer]
Hey! I added an example to the repo to demonstrate how you might do this: <https://github.com/actix/actix-web/blob/master/actix-files/examples/guarded-listing.rs>

The trick is in this setup, which is a bit of duplication but it's certainly the cleanest way:

```rust
.service(
    Files::new("/assets", EXAMPLES_DIR)
        .show_files_listing()
        .guard(guard::Header("show-listing", "?1")),
)
.service(Files::new("/assets", EXAMPLES_DIR))
```

(Sorry for the delay.)

**Comment by Necoo33** [comment]
> Hey! I added an example to the repo to demonstrate how you might do this: https://github.com/actix/actix-web/blob/master/actix-files/examples/guarded-listing.rs
> 
> The trick is in this setup, which is a bit of duplication but it's certainly the cleanest way:
> 
> ```rust
> .service(
>     Files::new("/assets", EXAMPLES_DIR)
>         .show_files_listing()
>         .guard(guard::Header("show-listing", "?1")),
> )
> .service(Files::new("/assets", EXAMPLES_DIR))
> ```
> 
> (Sorry for the delay.)

Thanks, it worked. When you don't answered yet i used a work around for kind of achieving this: basically i gave the first parameter of the `Files::new()` method's first argument to some hard to predict route and by that it became protected. By the way if you do this when you want to use these assets you have to import assets as that synthax on html pages: "someAssetRoute/yourAsset.png" that asset route should be the same with `Files::new()` method's first argument
